## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 501901`
Commit: `3efb481953c603a2fa75864f171647f58cbd5544`

---
### `x86_64-linux`
<details>
  <summary>:x: 59 packages failed to build:</summary>
  <ul>
    <li>aider-chat-full</li>
    <li>aider-chat-full.dist</li>
    <li>aider-chat-with-help</li>
    <li>aider-chat-with-help.dist</li>
    <li>alpaca</li>
    <li>ieda</li>
    <li>libretranslate (python313Packages.libretranslate)</li>
    <li>libretranslate.dist (python313Packages.libretranslate.dist)</li>
    <li>newelle</li>
    <li>python313Packages.amd-quark</li>
    <li>python313Packages.amd-quark.dist</li>
    <li>python313Packages.argos-translate-files</li>
    <li>python313Packages.argos-translate-files.dist</li>
    <li>python313Packages.argostranslate</li>
    <li>python313Packages.argostranslate.dist</li>
    <li>python313Packages.bambi</li>
    <li>python313Packages.bambi.dist</li>
    <li>python313Packages.brax</li>
    <li>python313Packages.brax.dist</li>
    <li>python313Packages.clu</li>
    <li>python313Packages.clu.dist</li>
    <li>python313Packages.distrax</li>
    <li>python313Packages.distrax.dist</li>
    <li>python313Packages.dm-sonnet</li>
    <li>python313Packages.dm-sonnet.dist</li>
    <li>python313Packages.lerobot</li>
    <li>python313Packages.lerobot.dist</li>
    <li>python313Packages.llama-index</li>
    <li>python313Packages.llama-index-cli</li>
    <li>python313Packages.llama-index-cli.dist</li>
    <li>python313Packages.llama-index-embeddings-huggingface</li>
    <li>python313Packages.llama-index-embeddings-huggingface.dist</li>
    <li>python313Packages.llama-index-vector-stores-chroma</li>
    <li>python313Packages.llama-index-vector-stores-chroma.dist</li>
    <li>python313Packages.llama-index.dist</li>
    <li>python313Packages.mmcv</li>
    <li>python313Packages.mmcv.dist</li>
    <li>python313Packages.onnx-asr</li>
    <li>python313Packages.onnx-asr.dist</li>
    <li>python313Packages.onnxscript</li>
    <li>python313Packages.onnxscript.dist</li>
    <li>python313Packages.rlax</li>
    <li>python313Packages.rlax.dist</li>
    <li>python313Packages.skrl</li>
    <li>python313Packages.skrl.dist</li>
    <li>python313Packages.translatehtml</li>
    <li>python313Packages.translatehtml.dist</li>
    <li>python313Packages.unsloth</li>
    <li>python313Packages.unsloth-zoo</li>
    <li>python313Packages.unsloth-zoo.dist</li>
    <li>python313Packages.unsloth.dist</li>
    <li>python314Packages.amd-quark</li>
    <li>python314Packages.amd-quark.dist</li>
    <li>python314Packages.mmcv</li>
    <li>python314Packages.mmcv.dist</li>
    <li>python314Packages.onnx-asr</li>
    <li>python314Packages.onnx-asr.dist</li>
    <li>python314Packages.onnxscript</li>
    <li>python314Packages.onnxscript.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 262 packages built:</summary>
  <ul>
    <li>adoptopenjdk-icedtea-web</li>
    <li>aitrack</li>
    <li>babeldoc</li>
    <li>babeldoc.dist</li>
    <li>calibre</li>
    <li>calibre-no-speech</li>
    <li>convertx</li>
    <li>crewai (python313Packages.crewai)</li>
    <li>crewai.dist (python313Packages.crewai.dist)</li>
    <li>crow-translate</li>
    <li>deface</li>
    <li>deface.dist</li>
    <li>envision</li>
    <li>exo</li>
    <li>exo.dist</li>
    <li>firefox-beta-unwrapped</li>
    <li>firefox-beta-unwrapped.debug</li>
    <li>firefox-beta-unwrapped.symbols</li>
    <li>firefox-devedition-unwrapped</li>
    <li>firefox-devedition-unwrapped.debug</li>
    <li>firefox-devedition-unwrapped.symbols</li>
    <li>firefox-unwrapped</li>
    <li>firefox-unwrapped.debug</li>
    <li>firefox-unwrapped.symbols</li>
    <li>firefoxpwa-unwrapped</li>
    <li>fotema</li>
    <li>frigate</li>
    <li>hieroglyphic</li>
    <li>hns</li>
    <li>hns.dist</li>
    <li>hyprwhspr-rs</li>
    <li>immich-machine-learning</li>
    <li>immich-machine-learning.dist</li>
    <li>kikoplay</li>
    <li>lacus</li>
    <li>lacus.dist</li>
    <li>librewolf-unwrapped</li>
    <li>librewolf-unwrapped.debug</li>
    <li>livecaptions</li>
    <li>maa-assistant-arknights</li>
    <li>maa-cli</li>
    <li>magika (python313Packages.magika)</li>
    <li>magika-cli</li>
    <li>magika.dist (python313Packages.magika.dist)</li>
    <li>markitdown-mcp</li>
    <li>markitdown-mcp.dist</li>
    <li>minari (python313Packages.minari)</li>
    <li>minari.dist (python313Packages.minari.dist)</li>
    <li>monado</li>
    <li>obs-studio-plugins.obs-backgroundremoval</li>
    <li>onnxruntime</li>
    <li>onnxruntime.debug</li>
    <li>onnxruntime.dev</li>
    <li>onnxruntime.dist</li>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>openseeface</li>
    <li>opentrack</li>
    <li>owocr</li>
    <li>owocr.dist</li>
    <li>piper-phonemize</li>
    <li>piper-tts</li>
    <li>piper-tts.dist</li>
    <li>python313Packages.ale-py</li>
    <li>python313Packages.ale-py.dist</li>
    <li>python313Packages.arviz</li>
    <li>python313Packages.arviz.dist</li>
    <li>python313Packages.chromadb</li>
    <li>python313Packages.chromadb.dist</li>
    <li>python313Packages.corner</li>
    <li>python313Packages.corner.dist</li>
    <li>python313Packages.deepface</li>
    <li>python313Packages.deepface.dist</li>
    <li>python313Packages.edward</li>
    <li>python313Packages.edward.dist</li>
    <li>python313Packages.evosax</li>
    <li>python313Packages.evosax.dist</li>
    <li>python313Packages.fastembed</li>
    <li>python313Packages.fastembed.dist</li>
    <li>python313Packages.faster-whisper</li>
    <li>python313Packages.faster-whisper.dist</li>
    <li>python313Packages.flax</li>
    <li>python313Packages.flax.dist</li>
    <li>python313Packages.flowjax</li>
    <li>python313Packages.flowjax.dist</li>
    <li>python313Packages.gliner</li>
    <li>python313Packages.gliner.dist</li>
    <li>python313Packages.gymnasium</li>
    <li>python313Packages.gymnasium.dist</li>
    <li>python313Packages.helium</li>
    <li>python313Packages.helium.dist</li>
    <li>python313Packages.insightface</li>
    <li>python313Packages.insightface.dist</li>
    <li>python313Packages.keras</li>
    <li>python313Packages.keras.dist</li>
    <li>python313Packages.lacuscore</li>
    <li>python313Packages.lacuscore.dist</li>
    <li>python313Packages.langchain-chroma</li>
    <li>python313Packages.langchain-chroma.dist</li>
    <li>python313Packages.langchain-huggingface</li>
    <li>python313Packages.langchain-huggingface.dist</li>
    <li>python313Packages.llm-sentence-transformers</li>
    <li>python313Packages.llm-sentence-transformers.dist</li>
    <li>python313Packages.manifest-ml</li>
    <li>python313Packages.manifest-ml.dist</li>
    <li>python313Packages.markitdown</li>
    <li>python313Packages.markitdown.dist</li>
    <li>python313Packages.mhcflurry</li>
    <li>python313Packages.mhcflurry.dist</li>
    <li>python313Packages.minisbd</li>
    <li>python313Packages.minisbd.dist</li>
    <li>python313Packages.mtcnn</li>
    <li>python313Packages.mtcnn.dist</li>
    <li>python313Packages.numpyro</li>
    <li>python313Packages.numpyro.dist</li>
    <li>python313Packages.nutpie</li>
    <li>python313Packages.nutpie.dist</li>
    <li>python313Packages.onnx-ir</li>
    <li>python313Packages.onnx-ir.dist</li>
    <li>python313Packages.onnxruntime</li>
    <li>python313Packages.onnxruntime.dist</li>
    <li>python313Packages.pettingzoo</li>
    <li>python313Packages.pettingzoo.dist</li>
    <li>python313Packages.piper-phonemize</li>
    <li>python313Packages.piper-phonemize.dist</li>
    <li>python313Packages.pymc</li>
    <li>python313Packages.pymc.dist</li>
    <li>python313Packages.pypasser</li>
    <li>python313Packages.pypasser.dist</li>
    <li>python313Packages.pytensor</li>
    <li>python313Packages.pytensor.dist</li>
    <li>python313Packages.rapidocr</li>
    <li>python313Packages.rapidocr-onnxruntime</li>
    <li>python313Packages.rapidocr-onnxruntime.dist</li>
    <li>python313Packages.rapidocr.dist</li>
    <li>python313Packages.rembg</li>
    <li>python313Packages.rembg.dist</li>
    <li>python313Packages.retinaface</li>
    <li>python313Packages.retinaface.dist</li>
    <li>python313Packages.sentence-transformers</li>
    <li>python313Packages.sentence-transformers.dist</li>
    <li>python313Packages.sherpa-onnx</li>
    <li>python313Packages.shimmy</li>
    <li>python313Packages.shimmy.dist</li>
    <li>python313Packages.speechrecognition</li>
    <li>python313Packages.speechrecognition.dist</li>
    <li>python313Packages.stable-baselines3</li>
    <li>python313Packages.stable-baselines3.dist</li>
    <li>python313Packages.tami4edgeapi</li>
    <li>python313Packages.tami4edgeapi.dist</li>
    <li>python313Packages.tensorflow-probability</li>
    <li>python313Packages.tensorflow-probability.dist</li>
    <li>python313Packages.tf2onnx</li>
    <li>python313Packages.tf2onnx.dist</li>
    <li>python313Packages.tianshou</li>
    <li>python313Packages.tianshou.dist</li>
    <li>python313Packages.tinygrad</li>
    <li>python313Packages.tinygrad.dist</li>
    <li>python313Packages.torchrl</li>
    <li>python313Packages.torchrl.dist</li>
    <li>python313Packages.torchtitan</li>
    <li>python313Packages.torchtitan.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
    <li>python313Packages.tyro</li>
    <li>python313Packages.tyro.dist</li>
    <li>ultralytics (python313Packages.ultralytics)</li>
    <li>ultralytics.dist (python313Packages.ultralytics.dist)</li>
    <li>python313Packages.unstructured-inference</li>
    <li>python313Packages.unstructured-inference.dist</li>
    <li>python313Packages.vmas</li>
    <li>python313Packages.vmas.dist</li>
    <li>python313Packages.vqgan-jax</li>
    <li>python313Packages.vqgan-jax.dist</li>
    <li>python313Packages.waymax</li>
    <li>python313Packages.waymax.dist</li>
    <li>python314Packages.chromadb</li>
    <li>python314Packages.chromadb.dist</li>
    <li>python314Packages.crewai</li>
    <li>python314Packages.crewai.dist</li>
    <li>python314Packages.fastembed</li>
    <li>python314Packages.fastembed.dist</li>
    <li>python314Packages.faster-whisper</li>
    <li>python314Packages.faster-whisper.dist</li>
    <li>python314Packages.gliner</li>
    <li>python314Packages.gliner.dist</li>
    <li>python314Packages.helium</li>
    <li>python314Packages.helium.dist</li>
    <li>python314Packages.insightface</li>
    <li>python314Packages.insightface.dist</li>
    <li>python314Packages.lacuscore</li>
    <li>python314Packages.lacuscore.dist</li>
    <li>python314Packages.langchain-chroma</li>
    <li>python314Packages.langchain-chroma.dist</li>
    <li>python314Packages.langchain-huggingface</li>
    <li>python314Packages.langchain-huggingface.dist</li>
    <li>python314Packages.llm-sentence-transformers</li>
    <li>python314Packages.llm-sentence-transformers.dist</li>
    <li>python314Packages.magika</li>
    <li>python314Packages.magika.dist</li>
    <li>python314Packages.manifest-ml</li>
    <li>python314Packages.manifest-ml.dist</li>
    <li>python314Packages.markitdown</li>
    <li>python314Packages.markitdown.dist</li>
    <li>python314Packages.minisbd</li>
    <li>python314Packages.minisbd.dist</li>
    <li>python314Packages.onnx-ir</li>
    <li>python314Packages.onnx-ir.dist</li>
    <li>python314Packages.onnxruntime</li>
    <li>python314Packages.onnxruntime.dist</li>
    <li>python314Packages.piper-phonemize</li>
    <li>python314Packages.piper-phonemize.dist</li>
    <li>python314Packages.pypasser</li>
    <li>python314Packages.pypasser.dist</li>
    <li>python314Packages.rapidocr</li>
    <li>python314Packages.rapidocr-onnxruntime</li>
    <li>python314Packages.rapidocr-onnxruntime.dist</li>
    <li>python314Packages.rapidocr.dist</li>
    <li>python314Packages.rembg</li>
    <li>python314Packages.rembg.dist</li>
    <li>python314Packages.sentence-transformers</li>
    <li>python314Packages.sentence-transformers.dist</li>
    <li>python314Packages.sherpa-onnx</li>
    <li>python314Packages.speechrecognition</li>
    <li>python314Packages.speechrecognition.dist</li>
    <li>python314Packages.tami4edgeapi</li>
    <li>python314Packages.tami4edgeapi.dist</li>
    <li>python314Packages.tinygrad</li>
    <li>python314Packages.tinygrad.dist</li>
    <li>python314Packages.txtai</li>
    <li>python314Packages.txtai.dist</li>
    <li>python314Packages.ultralytics</li>
    <li>python314Packages.ultralytics.dist</li>
    <li>python314Packages.unstructured-inference</li>
    <li>python314Packages.unstructured-inference.dist</li>
    <li>rabbit-ng</li>
    <li>rabbit-ng.dist</li>
    <li>rapidraw</li>
    <li>rembg</li>
    <li>rembg.dist</li>
    <li>seagoat</li>
    <li>seagoat.dist</li>
    <li>sherpa-onnx</li>
    <li>sherpa-onnx.debug</li>
    <li>sherpa-onnx.python</li>
    <li>sitespeed-io</li>
    <li>tests.home-assistant-component-tests.tami4</li>
    <li>thunderbird-unwrapped (thunderbird-latest-unwrapped, thunderbirdPackages.thunderbird, thunderbirdPackages.thunderbird-latest)</li>
    <li>thunderbird-unwrapped.debug (thunderbird-latest-unwrapped.debug, thunderbirdPackages.thunderbird-latest.debug, thunderbirdPackages.thunderbird.debug)</li>
    <li>thunderbird-unwrapped.symbols (thunderbird-latest-unwrapped.symbols, thunderbirdPackages.thunderbird-latest.symbols, thunderbirdPackages.thunderbird.symbols)</li>
    <li>unbook</li>
    <li>unstructured-api</li>
    <li>vcmi</li>
    <li>vectorcode</li>
    <li>vectorcode.dist</li>
    <li>vimPlugins.vectorcode-nvim</li>
    <li>voxtype-onnx</li>
    <li>vtk_9_6</li>
    <li>wyoming-faster-whisper</li>
    <li>wyoming-faster-whisper.dist</li>
    <li>wyoming-piper</li>
    <li>wyoming-piper.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>ieda</summary>
<pre>[1226/1238] Building CXX object src/operation/iPA/CMakeFiles/iPower.dir/main.cc.o
warning: Skipping impure flag -march=native because NIX_ENFORCE_NO_NATIVE is set
In file included from /build/iEDA-src-0.1.0-unstable-2025-12-23/src/operation/iPA/source/module/core/PwrVertex.hh:29,
                 from /build/iEDA-src-0.1.0-unstable-2025-12-23/src/operation/iPA/source/module/core/PwrArc.hh:27,
                 from /build/iEDA-src-0.1.0-unstable-2025-12-23/src/operation/iPA/source/module/core/PwrGraph.hh:30,
                 from /build/iEDA-src-0.1.0-unstable-2025-12-23/src/operation/iPA/api/Power.hh:28,
                 from /build/iEDA-src-0.1.0-unstable-2025-12-23/src/operation/iPA/main.cc:29:
/build/iEDA-src-0.1.0-unstable-2025-12-23/src/operation/iPA/source/module/core/PwrData.hh:57:20: warning: &#x27;virtual unsigned int ipower::PwrData::compareSignature(const ipower::PwrData*) const&#x27; was hidden [-Woverloaded-virtual=]
   57 |   virtual unsigned compareSignature(const PwrData* data) const {
      |                    ^~~~~~~~~~~~~~~~
/build/iEDA-src-0.1.0-unstable-2025-12-23/src/operation/iPA/source/module/core/PwrData.hh:92:12: note:   by &#x27;unsigned int ipower::PwrSPData::compareSignature(const ipower::PwrSPData*) const&#x27;
   92 |   unsigned compareSignature(const PwrSPData* data) const {
      |            ^~~~~~~~~~~~~~~~
/build/iEDA-src-0.1.0-unstable-2025-12-23/src/operation/iPA/source/module/core/PwrData.hh:57:20: warning: &#x27;virtual unsigned int ipower::PwrData::compareSignature(const ipower::PwrData*) const&#x27; was hidden [-Woverloaded-virtual=]
   57 |   virtual unsigned compareSignature(const PwrData* data) const {
      |                    ^~~~~~~~~~~~~~~~
/build/iEDA-src-0.1.0-unstable-2025-12-23/src/operation/iPA/source/module/core/PwrData.hh:133:12: note:   by &#x27;unsigned int ipower::PwrToggleData::compareSignature(const ipower::PwrToggleData*) const&#x27;
  133 |   unsigned compareSignature(const PwrToggleData* data) const {
      |            ^~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.</pre>
</details>
<details>
<summary>python313Packages.distrax</summary>
<pre>distrax/_src/utils/conversion_test.py::AsFloatArrayTest::test_on_invalid_array3
  /build/source/distrax/_src/utils/conversion_test.py:175: UserWarning: Explicitly requested dtype complex128 requested in zeros is not available, and will be truncated to dtype complex64. To enable more dtypes, set the jax_enable_x64 configuration option or the JAX_ENABLE_X64 shell environment variable. See https://github.com/jax-ml/jax#current-gotchas for more.
    x = jnp.zeros([], dtype)

distrax/_src/utils/math_test.py::MathTest::test_multiply_no_nan_grads
  /build/source/distrax/_src/utils/math.py:75: DeprecationWarning: jax.core.get_aval is deprecated; use jax.typeof instead.
    primal_aval = jax_core.get_aval(primal_out)

distrax/_src/distributions/von_mises_test.py::VonMisesTest::test_von_mises_log_cdf
  /build/source/distrax/_src/distributions/von_mises_test.py:320: RuntimeWarning: divide by zero encountered in log
    np.testing.assert_allclose(np.log(cdf), logcdf, atol=1e-4, rtol=1e-4)

distrax/_src/distributions/von_mises_test.py::VonMisesTest::test_von_mises_log_survival
  /build/source/distrax/_src/distributions/von_mises_test.py:329: RuntimeWarning: divide by zero encountered in log
    np.testing.assert_allclose(np.log(surv), logsurv, atol=1e-4, rtol=1e-4)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED distrax/_src/bijectors/diag_plus_low_rank_linear_test.py::DiagPlusLowRankLinearTest::test_raises_on_invalid_input_shape - execnet.gateway_base.DumpError: can&#x27;t serialize &lt;class &#x27;method&#x27;&gt;
==== 1 failed, 5714 passed, 1025 skipped, 2839 warnings in 73.46s (0:01:13) ====</pre>
</details>
<details>
<summary>python313Packages.onnxscript</summary>
<pre>
@script()
def bck_test_unsqueeze_two_axes(x: FLOAT[3,4,5], axes: INT64[2]) -&gt; (FLOAT[3,1,4,5,1]):
    y = opset25.Unsqueeze(x, axes)
    return y
FAILED onnxscript/backend/onnx_export_test.py::TestOnnxBackEnd::test_export2python_produces_correct_onnx_script_model_1626_test_transpose_all_permutations_3 - AssertionError: Unable to import &#x27;tests.onnx_backend_test_code.test_transpose_all_permutations_3&#x27; (e=cannot import name &#x27;opset25&#x27; from &#x27;onnxscript.onnx_opset&#x27; (/build/source/onnxscript/onnx_opset/__init__.py)) (file: &#x27;/build/source/tests/onnx_backend_test_code/test_transpose_all_permutations_3.py&#x27;, absolute path: &#x27;/build/source/tests/onnx_backend_test_code/test_transpose_all_permutations_3.py&#x27;, current folder: /build/source
---- CONTENT --
import numpy as np
from onnx import TensorProto
from onnx.helper import make_tensor
from onnxscript import script, external_tensor
from onnxscript.values import Opset
from onnxscript.onnx_types import FLOAT
from onnxscript.onnx_opset import opset25

@script()
def bck_test_transpose_all_permutations_3(data: FLOAT[2,3,4]) -&gt; (FLOAT[3,4,2]):
    transposed = opset25.Transpose(data, perm=[1, 2, 0])
    return transposed
= 88 failed, 1964 passed, 490 skipped, 5 deselected, 5 xfailed, 17450 warnings in 495.67s (0:08:15) =</pre>
</details>
<details>
<summary>python313Packages.unsloth-zoo</summary>
<pre>  File &quot;&lt;frozen importlib._bootstrap_external&gt;&quot;, line 1023, in exec_module
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 488, in _call_with_frames_removed
  File &quot;/nix/store/qnbg1bcdai60n5h2vmjcl3b7s6gwym42-python3.13-unsloth-zoo-2026.3.4/lib/python3.13/site-packages/unsloth_zoo/__init__.py&quot;, line 284, in &lt;module&gt;
    from .temporary_patches import (
        encode_conversations_with_harmony,
    )
  File &quot;/nix/store/qnbg1bcdai60n5h2vmjcl3b7s6gwym42-python3.13-unsloth-zoo-2026.3.4/lib/python3.13/site-packages/unsloth_zoo/temporary_patches/__init__.py&quot;, line 22, in &lt;module&gt;
    from .gpt_oss import *
  File &quot;/nix/store/qnbg1bcdai60n5h2vmjcl3b7s6gwym42-python3.13-unsloth-zoo-2026.3.4/lib/python3.13/site-packages/unsloth_zoo/temporary_patches/gpt_oss.py&quot;, line 1125, in &lt;module&gt;
    device_memory = torch.cuda.memory.mem_get_info(0)[-1]
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
  File &quot;/nix/store/0gf9b8xznpzn5r5aml9syxma7519xv7r-python3.13-torch-2.10.0/lib/python3.13/site-packages/torch/cuda/memory.py&quot;, line 897, in mem_get_info
    return torch.cuda.cudart().cudaMemGetInfo(device)
           ~~~~~~~~~~~~~~~~~^^
  File &quot;/nix/store/0gf9b8xznpzn5r5aml9syxma7519xv7r-python3.13-torch-2.10.0/lib/python3.13/site-packages/torch/cuda/__init__.py&quot;, line 501, in cudart
    _lazy_init()
    ~~~~~~~~~~^^
  File &quot;/nix/store/0gf9b8xznpzn5r5aml9syxma7519xv7r-python3.13-torch-2.10.0/lib/python3.13/site-packages/torch/cuda/__init__.py&quot;, line 417, in _lazy_init
    raise AssertionError(&quot;Torch not compiled with CUDA enabled&quot;)
AssertionError: Torch not compiled with CUDA enabled</pre>
</details>
<details>
<summary>python314Packages.onnxscript</summary>
<pre>
@script()
def bck_test_reshape_allowzero_reordered(data: FLOAT[0,3,4], shape: INT64[3]) -&gt; (FLOAT[3,4,0]):
    reshaped = opset25.Reshape(data, shape, allowzero=1)
    return reshaped
FAILED onnxscript/backend/onnx_export_test.py::TestOnnxBackEnd::test_export2python_produces_correct_onnx_script_model_1645_test_castlike_DOUBLE_to_FLOAT16 - AssertionError: Unable to import &#x27;tests.onnx_backend_test_code.test_castlike_DOUBLE_to_FLOAT16&#x27; (e=cannot import name &#x27;opset25&#x27; from &#x27;onnxscript.onnx_opset&#x27; (/build/source/onnxscript/onnx_opset/__init__.py)) (file: &#x27;/build/source/tests/onnx_backend_test_code/test_castlike_DOUBLE_to_FLOAT16.py&#x27;, absolute path: &#x27;/build/source/tests/onnx_backend_test_code/test_castlike_DOUBLE_to_FLOAT16.py&#x27;, current folder: /build/source
---- CONTENT --
import numpy as np
from onnx import TensorProto
from onnx.helper import make_tensor
from onnxscript import script, external_tensor
from onnxscript.values import Opset
from onnxscript.onnx_types import DOUBLE, FLOAT16
from onnxscript.onnx_opset import opset25

@script()
def bck_test_castlike_DOUBLE_to_FLOAT16(input: DOUBLE[3,4], like: FLOAT16[0]) -&gt; (FLOAT16[3,4]):
    output = opset25.CastLike(input, like)
    return output
= 88 failed, 1964 passed, 490 skipped, 5 deselected, 5 xfailed, 17450 warnings in 617.91s (0:10:17) =</pre>
</details>
</details>

--- a/pkgs/by-name/on/onnxruntime/package.nix
+++ b/pkgs/by-name/on/onnxruntime/package.nix
@@ -3,10 +3,7 @@
   config,
   stdenv,
   fetchFromGitHub,
-  applyPatches,
-  fetchpatch,
-  fetchurl,
-  abseil-cpp_202407,
+  abseil-cpp_202508,
   cmake,
   cpuinfo,
   eigen,
@@ -59,39 +56,20 @@ let
     hash = "sha256-pjwjrqq6dfiVsXIhbBtbolhiysiFlFTnx5XcX77f+C0=";
   };
 
-  onnx-src = applyPatches {
+  onnx-src = fetchFromGitHub {
     name = "onnx-src";
-    src = fetchFromGitHub {
-      owner = "onnx";
-      repo = "onnx";
-      tag = "v1.18.0";
-      hash = "sha256-UhtF+CWuyv5/Pq/5agLL4Y95YNP63W2BraprhRqJOag=";
-    };
-
-    patches = [
-      # Fix "error: conversion from 'onnx::OpSchema' to non-scalar type 'onnx::OpSchemaRegistry::OpSchemaRegisterOnce'"
-      # https://github.com/microsoft/onnxruntime/issues/26229
-      # Fix from https://github.com/onnx/onnx/pull/7390
-      (fetchpatch {
-        url = "https://github.com/onnx/onnx/commit/595a069aaac07586f111681245bc808ee63551f8.patch";
-        includes = [ "onnx/defs/schema.h" ];
-        hash = "sha256-FFAJuJse4nmNT3ixvEdlqzbr3edY46SqEFv7z/oo6m0=";
-      })
-
-      # Fix "undefined reference to `onnx::RNNShapeInference(onnx::InferenceContext&)'"
-      (fetchpatch {
-        url = "https://github.com/onnx/onnx/commit/6769c41ad64ebca0358da8c7211d2c6d0e627b2b.patch";
-        hash = "sha256-VlTHs0om20kTNvSVQaasSsa5JROliQy4k9BECTsBtbU=";
-      })
-    ];
+    owner = "onnx";
+    repo = "onnx";
+    tag = "v1.20.1";
+    hash = "sha256-XZJXD6sBvVJ6cLPyDkKOW8oSkjqcw9whUqDWd7dxY3c=";
   };
 
   cutlass-src = fetchFromGitHub {
     name = "cutlass-src";
     owner = "NVIDIA";
     repo = "cutlass";
-    tag = "v3.9.2";
-    hash = "sha256-teziPNA9csYvhkG5t2ht8W8x5+1YGGbHm8VKx4JoxgI=";
+    tag = "v4.2.1";
+    hash = "sha256-iP560D5Vwuj6wX1otJhwbvqe/X4mYVeKTpK533Wr5gY=";
   };
 
   dlpack-src = fetchFromGitHub {
@@ -106,30 +84,17 @@ let
 in
 effectiveStdenv.mkDerivation (finalAttrs: {
   pname = "onnxruntime";
-  version = "1.23.2";
+  version = "1.24.4";
 
   src = fetchFromGitHub {
     owner = "microsoft";
     repo = "onnxruntime";
     tag = "v${finalAttrs.version}";
     fetchSubmodules = true;
-    hash = "sha256-hZ2L5+0Enkw4rGDKVpRECnKXP87w6Kbiyp6Fdxwt6hk=";
+    hash = "sha256-CjPgRkPyp7dUPAOo3cePWQvucOlQAwtT4NO5w3NkV+E=";
   };
 
   patches = [
-    # Missing cstdint include (GCC 15 compatibility)
-    (fetchpatch {
-      url = "https://github.com/microsoft/onnxruntime/commit/d6e712c5b7b6260a61e54d1fe40107cf5366ee77.patch";
-      hash = "sha256-FSuPybX8f2VoxvLhcYx4rdChaiK8bSUDR32sN3Efwfc=";
-    })
-
-    # Correct maybe-uninitialized and range-loop-construct warnings
-    # https://github.com/microsoft/onnxruntime/pull/26201
-    (fetchpatch {
-      url = "https://github.com/microsoft/onnxruntime/commit/8ebd0bf1cf02414584d15d7244b07fa97d65ba02.patch";
-      hash = "sha256-vX+kaFiNdmqWI91JELcLpoaVIHBb5EPbI7rCAMYAx04=";
-    })
-
     # Skip execinfo include on musl
     # https://github.com/microsoft/onnxruntime/pull/25726
     ./musl-execinfo.patch
@@ -139,15 +104,35 @@ effectiveStdenv.mkDerivation (finalAttrs: {
 
     # Fix build of unit tests with musl libc
     # https://github.com/microsoft/onnxruntime/issues/9155
-    (fetchurl {
-      url = "https://gitlab.alpinelinux.org/alpine/aports/-/raw/462dfe0eb4b66948fe48de44545cc22bb64fdf9f/community/onnxruntime/0001-Remove-MATH_NO_EXCEPT-macro.patch";
-      hash = "sha256-BdeGYevZExWWCuJ1lSw0Roy3h+9EbJgFF8qMwVxSn1A=";
-    })
+    # Patch adapted from https://gitlab.alpinelinux.org/alpine/aports/-/raw/462dfe0eb4b66948fe48de44545cc22bb64fdf9f/community/onnxruntime/0001-Remove-MATH_NO_EXCEPT-macro.patch
+    ./remove-MATH_NO_EXCEPT-macro.patch
 
     # Fix build of ignored outputs after Protobuf 34 added `[[nodiscard]]` to
     # many functions.
     ./protobuf34-nodiscard.patch
   ];
+
+  postPatch = ''
+    substituteInPlace cmake/libonnxruntime.pc.cmake.in \
+      --replace-fail '$'{prefix}/@CMAKE_INSTALL_ @CMAKE_INSTALL_
+    echo "find_package(cudnn_frontend REQUIRED)" > cmake/external/cudnn_frontend.cmake
+  ''
+  + ''
+    substituteInPlace onnxruntime/core/platform/posix/env.cc --replace-fail \
+      "return PathString{};" \
+      "return PathString(\"$out/lib/\");"
+  ''
+  + lib.optionalString rocmSupport ''
+    patchShebangs tools/ci_build/hipify-perl
+  ''
+  # https://github.com/NixOS/nixpkgs/pull/226734#issuecomment-1663028691
+  + lib.optionalString (effectiveStdenv.hostPlatform.system == "aarch64-linux") ''
+    rm -v onnxruntime/test/optimizer/nhwc_transformer_test.cc
+  '';
+
+  postBuild = lib.optionalString pythonSupport ''
+    ${python3Packages.python.interpreter} ../setup.py bdist_wheel
+  '';
 
   nativeBuildInputs = [
     cmake
@@ -261,7 +246,7 @@ effectiveStdenv.mkDerivation (finalAttrs: {
     (lib.cmakeFeature "CMAKE_CXX_FLAGS" "-Wno-error=unused-variable")
     (lib.cmakeBool "FETCHCONTENT_FULLY_DISCONNECTED" true)
     (lib.cmakeBool "FETCHCONTENT_QUIET" false)
-    (lib.cmakeFeature "FETCHCONTENT_SOURCE_DIR_ABSEIL_CPP" "${abseil-cpp_202407.src}")
+    (lib.cmakeFeature "FETCHCONTENT_SOURCE_DIR_ABSEIL_CPP" "${abseil-cpp_202508.src}")
     (lib.cmakeFeature "FETCHCONTENT_SOURCE_DIR_DLPACK" "${dlpack-src}")
     (lib.cmakeFeature "FETCHCONTENT_SOURCE_DIR_FLATBUFFERS" "${flatbuffers_23.src}")
     (lib.cmakeFeature "FETCHCONTENT_SOURCE_DIR_MP11" "${mp11-src}")
@@ -343,29 +328,6 @@ effectiveStdenv.mkDerivation (finalAttrs: {
     "nostrictaliasing"
   ];
   hardeningDisable = lib.optional effectiveStdenv.hostPlatform.isMusl "fortify";
-
-  postPatch = ''
-    substituteInPlace cmake/libonnxruntime.pc.cmake.in \
-      --replace-fail '$'{prefix}/@CMAKE_INSTALL_ @CMAKE_INSTALL_
-    echo "find_package(cudnn_frontend REQUIRED)" > cmake/external/cudnn_frontend.cmake
-  ''
-  # https://github.com/microsoft/onnxruntime/blob/c4f3742bb456a33ee9c826ce4e6939f8b84ce5b0/onnxruntime/core/platform/env.h#L249
-  + ''
-    substituteInPlace onnxruntime/core/platform/env.h --replace-fail \
-      "GetRuntimePath() const { return PathString(); }" \
-      "GetRuntimePath() const { return PathString(\"$out/lib/\"); }"
-  ''
-  + lib.optionalString rocmSupport ''
-    patchShebangs tools/ci_build/hipify-perl
-  ''
-  # https://github.com/NixOS/nixpkgs/pull/226734#issuecomment-1663028691
-  + lib.optionalString (effectiveStdenv.hostPlatform.system == "aarch64-linux") ''
-    rm -v onnxruntime/test/optimizer/nhwc_transformer_test.cc
-  '';
-
-  postBuild = lib.optionalString pythonSupport ''
-    ${python3Packages.python.interpreter} ../setup.py bdist_wheel
-  '';
 
   # perform parts of `tools/ci_build/github/linux/copy_strip_binary.sh`
   postInstall = ''

--- a/pkgs/by-name/on/onnxruntime/protobuf34-nodiscard.patch
+++ b/pkgs/by-name/on/onnxruntime/protobuf34-nodiscard.patch
@@ -1,8 +1,8 @@
 diff --git a/onnxruntime/core/framework/graph_partitioner.cc b/onnxruntime/core/framework/graph_partitioner.cc
-index 43caf4766d..fabc2a26a0 100644
+index 9cb2111670..8a3ec5bab9 100644
 --- a/onnxruntime/core/framework/graph_partitioner.cc
 +++ b/onnxruntime/core/framework/graph_partitioner.cc
-@@ -957,7 +957,7 @@
+@@ -966,7 +966,7 @@ static Status CreateEpContextModel(const ExecutionProviders& execution_providers
  
      AllocatorPtr allocator = output_buffer_holder->buffer_allocator;
      IAllocatorUniquePtr<void> buffer = IAllocator::MakeUniquePtr<void>(allocator, buffer_size);
@@ -11,7 +11,7 @@ index 43caf4766d..fabc2a26a0 100644
  
      *output_buffer_holder->buffer_size_ptr = buffer_size;
      *output_buffer_holder->buffer_ptr = buffer.release();
-@@ -970,7 +970,7 @@
+@@ -979,7 +979,7 @@ static Status CreateEpContextModel(const ExecutionProviders& execution_providers
      auto out_stream_buf = std::make_unique<epctx::OutStreamBuf>(*output_write_func_holder);
      std::ostream out_stream(out_stream_buf.get());
  
@@ -21,10 +21,10 @@ index 43caf4766d..fabc2a26a0 100644
      ORT_RETURN_IF_ERROR(out_stream_buf->GetStatus());
    } else {
 diff --git a/onnxruntime/python/onnxruntime_pybind_schema.cc b/onnxruntime/python/onnxruntime_pybind_schema.cc
-index cd1d2a8da1..05da1a1447 100644
+index 8cb617fe52..254fc1abf9 100644
 --- a/onnxruntime/python/onnxruntime_pybind_schema.cc
 +++ b/onnxruntime/python/onnxruntime_pybind_schema.cc
-@@ -169,7 +169,7 @@
+@@ -163,7 +163,7 @@ void addOpSchemaSubmodule(py::module& m) {
            "_default_value",
            [](ONNX_NAMESPACE::OpSchema::Attribute* attr) -> py::bytes {
              std::string out;
@@ -34,10 +34,10 @@ index cd1d2a8da1..05da1a1447 100644
            })
        .def_readonly("required", &ONNX_NAMESPACE::OpSchema::Attribute::required);
 diff --git a/onnxruntime/test/ep_graph/test_ep_graph.cc b/onnxruntime/test/ep_graph/test_ep_graph.cc
-index 7e6d157799..f08b09990d 100644
+index 055b255132..3925db8ac7 100644
 --- a/onnxruntime/test/ep_graph/test_ep_graph.cc
 +++ b/onnxruntime/test/ep_graph/test_ep_graph.cc
-@@ -232,7 +232,7 @@
+@@ -261,7 +261,7 @@ TEST(EpGraphTest, SerializeToProto_InputModelHasExternalIni) {
                                                          handle_initializer_data));
  
      std::ofstream ofs(serialized_model_path, std::ios::binary);
@@ -46,7 +46,7 @@ index 7e6d157799..f08b09990d 100644
      ofs.flush();
  
      ASSERT_TRUE(std::filesystem::exists(serialized_model_path));
-@@ -357,7 +357,7 @@
+@@ -395,7 +395,7 @@ TEST(EpGraphTest, SerializeToProto_Mnist) {
                                                          handle_initializer_data));
  
      std::ofstream ofs(serialized_model_path, std::ios::binary);
@@ -55,7 +55,7 @@ index 7e6d157799..f08b09990d 100644
      ofs.flush();
  
      ASSERT_TRUE(std::filesystem::exists(serialized_model_path));
-@@ -487,7 +487,7 @@
+@@ -525,7 +525,7 @@ TEST(EpGraphTest, SerializeToProto_ConstantOfShape) {
                                                          handle_initializer_data));
  
      std::ofstream ofs(serialized_model_path, std::ios::binary);
@@ -64,7 +64,7 @@ index 7e6d157799..f08b09990d 100644
      ofs.flush();
  
      ASSERT_TRUE(std::filesystem::exists(serialized_model_path));
-@@ -552,7 +552,7 @@
+@@ -590,7 +590,7 @@ TEST(EpGraphTest, SerializeToProto_3LayerSubgraphs) {
      ASSERT_CXX_ORTSTATUS_OK(OrtEpUtils::OrtGraphToProto(test_graph->GetOrtGraph(), model_proto));
  
      std::ofstream ofs(serialized_model_path, std::ios::binary);
@@ -73,24 +73,73 @@ index 7e6d157799..f08b09990d 100644
      ofs.flush();
  
      ASSERT_TRUE(std::filesystem::exists(serialized_model_path));
-diff --git a/onnxruntime/test/optimizer/graph_transform_test_builder.cc b/onnxruntime/test/optimizer/graph_transform_test_builder.cc
-index 756cc4159e..8323851385 100644
---- a/onnxruntime/test/optimizer/graph_transform_test_builder.cc
-+++ b/onnxruntime/test/optimizer/graph_transform_test_builder.cc
-@@ -158,7 +158,7 @@
+diff --git a/onnxruntime/test/framework/inference_session_test.cc b/onnxruntime/test/framework/inference_session_test.cc
+index 1c4e7800b7..8ceb05f0f6 100644
+--- a/onnxruntime/test/framework/inference_session_test.cc
++++ b/onnxruntime/test/framework/inference_session_test.cc
+@@ -1045,7 +1045,7 @@ static void TestBindHelper(const std::string& log_str,
+   CreateMatMulModel(p_model, run_provider_type);
  
-   // Serialize the model to a string.
-   std::string model_data;
--  model.ToProto().SerializeToString(&model_data);
-+  ASSERT_TRUE(model.ToProto().SerializeToString(&model_data));
-   std::shared_ptr<IExecutionProvider> ep_shared = ep ? std::move(ep) : nullptr;
+   std::string s1;
+-  p_model->ToProto().SerializeToString(&s1);
++  ASSERT_TRUE(p_model->ToProto().SerializeToString(&s1));
+   std::stringstream sstr(s1);
+   ASSERT_STATUS_OK(session_object.Load(sstr));
+   ASSERT_STATUS_OK(session_object.Initialize());
+@@ -1078,7 +1078,7 @@ TEST(InferenceSessionTests, TestIOBindingReuse) {
+   CreateMatMulModel(p_model, kCpuExecutionProvider);
  
-   auto run_model = [&](TransformerLevel level, std::vector<OrtValue>& fetches,
+   std::string s1;
+-  p_model->ToProto().SerializeToString(&s1);
++  ASSERT_TRUE(p_model->ToProto().SerializeToString(&s1));
+   std::stringstream sstr(s1);
+   ASSERT_TRUE(session_object.Load(sstr).IsOK());
+   ASSERT_STATUS_OK(session_object.Initialize());
+diff --git a/onnxruntime/test/ir/graph_test.cc b/onnxruntime/test/ir/graph_test.cc
+index 4d80cb7047..c69ac46c05 100644
+--- a/onnxruntime/test/ir/graph_test.cc
++++ b/onnxruntime/test/ir/graph_test.cc
+@@ -1244,7 +1244,7 @@ TEST_F(GraphTest, GraphConstruction_CheckGraphInputOutputOrderMaintained) {
+   auto proto = model.ToProto();
+   std::string s1;
+   // std::stringstream s1;
+-  model.ToProto().SerializeToString(&s1);
++  ASSERT_TRUE(model.ToProto().SerializeToString(&s1));
+ 
+   ModelProto model_proto;
+   //  const bool result = model_proto.ParseFromIstream(&s1);
+@@ -1313,7 +1313,7 @@ TEST_F(GraphTest, UnusedInitializerAndNodeArgsAreIgnored) {
+   auto proto = model.ToProto();
+   std::string s1;
+   // std::stringstream s1;
+-  model.ToProto().SerializeToString(&s1);
++  ASSERT_TRUE(model.ToProto().SerializeToString(&s1));
+ 
+   ModelProto model_proto;
+   const bool result = model_proto.ParseFromString(s1);
+@@ -1342,7 +1342,7 @@ TEST_F(GraphTest, UnusedSparseInitializerIsIgnored) {
+     ConstructASimpleAddGraph(*m_graph, nullptr);
+     auto* m_sparse_initializer = m_graph->add_sparse_initializer();
+     ConstructSparseTensor("unused_sparse_initializer", *m_sparse_initializer);
+-    model_proto.SerializeToString(&s1);
++    ASSERT_TRUE(model_proto.SerializeToString(&s1));
+   }
+ 
+   ModelProto model_proto_1;
+@@ -1940,7 +1940,7 @@ TEST_F(GraphTest, SparseInitializerHandling) {
+     ConstructASimpleAddGraph(*m_graph, nullptr);
+     auto* m_sparse_initializer = m_graph->add_sparse_initializer();
+     ConstructSparseTensor(input_initializer_name, *m_sparse_initializer);
+-    model_proto.SerializeToString(&s1);
++    ASSERT_TRUE(model_proto.SerializeToString(&s1));
+   }
+ 
+   ModelProto model_proto_sparse;
 diff --git a/onnxruntime/test/optimizer/nchwc_optimizer_test.cc b/onnxruntime/test/optimizer/nchwc_optimizer_test.cc
-index 538f600404..111692d4ca 100644
+index fc0ba86c7f..b086040212 100644
 --- a/onnxruntime/test/optimizer/nchwc_optimizer_test.cc
 +++ b/onnxruntime/test/optimizer/nchwc_optimizer_test.cc
-@@ -185,7 +185,7 @@
+@@ -185,7 +185,7 @@ void NchwcOptimizerTester(const std::function<void(NchwcTestHelper& helper)>& bu
  
    // Serialize the model to a string.
    std::string model_data;
@@ -99,3 +148,56 @@ index 538f600404..111692d4ca 100644
  
    auto run_model = [&](TransformerLevel level, std::vector<OrtValue>& fetches) {
      SessionOptions session_options;
+diff --git a/onnxruntime/test/providers/compare_provider_test_utils.cc b/onnxruntime/test/providers/compare_provider_test_utils.cc
+index 6312014387..578d72cb09 100644
+--- a/onnxruntime/test/providers/compare_provider_test_utils.cc
++++ b/onnxruntime/test/providers/compare_provider_test_utils.cc
+@@ -83,7 +83,7 @@ void CompareOpTester::CompareWithCPU(const std::string& target_provider_type,
+ 
+   // first run with cpu
+   std::string s1;
+-  model.ToProto().SerializeToString(&s1);
++  ASSERT_TRUE(model.ToProto().SerializeToString(&s1));
+   std::istringstream model_proto_str(s1);
+ 
+   ASSERT_STATUS_OK(cpu_session_object.Load(model_proto_str));
+@@ -104,7 +104,7 @@ void CompareOpTester::CompareWithCPU(const std::string& target_provider_type,
+   ASSERT_STATUS_OK(target_session_object.RegisterExecutionProvider(std::move(target_execution_provider)));
+ 
+   std::string s2;
+-  tp_model.ToProto().SerializeToString(&s2);
++  ASSERT_TRUE(tp_model.ToProto().SerializeToString(&s2));
+   std::istringstream model_proto_str1(s2);
+   ASSERT_STATUS_OK(target_session_object.Load(model_proto_str1));
+ 
+@@ -159,7 +159,7 @@ void CompareOpTester::CompareEPs(const std::shared_ptr<IExecutionProvider>& sour
+ 
+   // first run with source provider
+   std::string s1;
+-  model.ToProto().SerializeToString(&s1);
++  ASSERT_TRUE(model.ToProto().SerializeToString(&s1));
+   std::istringstream model_proto_str(s1);
+ 
+   ASSERT_STATUS_OK(source_session_object.Load(model_proto_str));
+@@ -181,7 +181,7 @@ void CompareOpTester::CompareEPs(const std::shared_ptr<IExecutionProvider>& sour
+     ASSERT_STATUS_OK(target_session_object.RegisterExecutionProvider(target_execution_provider));
+ 
+     std::string s2;
+-    tp_model.ToProto().SerializeToString(&s2);
++    ASSERT_TRUE(tp_model.ToProto().SerializeToString(&s2));
+     std::istringstream model_proto_str1(s2);
+     ASSERT_STATUS_OK(target_session_object.Load(model_proto_str1));
+ 
+diff --git a/onnxruntime/test/unittest_util/graph_transform_test_builder.cc b/onnxruntime/test/unittest_util/graph_transform_test_builder.cc
+index 5caafa0f37..327463c615 100644
+--- a/onnxruntime/test/unittest_util/graph_transform_test_builder.cc
++++ b/onnxruntime/test/unittest_util/graph_transform_test_builder.cc
+@@ -158,7 +158,7 @@ void TransformerTester(const std::function<void(ModelTestBuilder& helper)>& buil
+ 
+   // Serialize the model to a string.
+   std::string model_data;
+-  model.ToProto().SerializeToString(&model_data);
++  ASSERT_TRUE(model.ToProto().SerializeToString(&model_data));
+   std::shared_ptr<IExecutionProvider> ep_shared = ep ? std::move(ep) : nullptr;
+ 
+   auto run_model = [&](TransformerLevel level, std::vector<OrtValue>& fetches,

--- a/pkgs/by-name/on/onnxruntime/remove-MATH_NO_EXCEPT-macro.patch
+++ b/pkgs/by-name/on/onnxruntime/remove-MATH_NO_EXCEPT-macro.patch
@@ -1,0 +1,37 @@
+diff --git a/onnxruntime/test/providers/cpu/math/element_wise_ops_test.cc b/onnxruntime/test/providers/cpu/math/element_wise_ops_test.cc
+index 3fb8cc3e15..ccc9bd543e 100644
+--- a/onnxruntime/test/providers/cpu/math/element_wise_ops_test.cc
++++ b/onnxruntime/test/providers/cpu/math/element_wise_ops_test.cc
+@@ -3797,13 +3797,7 @@ TEST(MathOpTest, Mean_8) {
+   test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kTensorrtExecutionProvider});  // TensorRT: Input batch size is inconsistent
+ }
+ 
+-#ifdef _LIBCPP_VERSION
+-#define MATH_NO_EXCEPT
+-#else
+-#define MATH_NO_EXCEPT noexcept
+-#endif
+-
+-template <float (&op)(float value) MATH_NO_EXCEPT>
++template <float (&op)(float value)>
+ void TrigFloatTest(OpTester& test, std::initializer_list<float> input, float abs_error = -1.0f) {
+   std::vector<int64_t> dims{static_cast<int64_t>(input.size())};
+ 
+@@ -3821,7 +3815,7 @@ void TrigFloatTest(OpTester& test, std::initializer_list<float> input, float abs
+   test.Run();
+ }
+ 
+-template <double (&op)(double value) MATH_NO_EXCEPT>
++template <double (&op)(double value)>
+ void TrigDoubleTest(OpTester& test, std::initializer_list<double> input,
+                     const std::unordered_set<std::string> excluded_provider_types = {}) {
+   std::vector<int64_t> dims{static_cast<int64_t>(input.size())};
+@@ -3835,7 +3829,7 @@ void TrigDoubleTest(OpTester& test, std::initializer_list<double> input,
+   test.Run(OpTester::ExpectResult::kExpectSuccess, "", excluded_provider_types);
+ }
+ 
+-template <float (&op)(float value) MATH_NO_EXCEPT>
++template <float (&op)(float value)>
+ void TrigFloat16Test(OpTester& test, std::initializer_list<float> input) {
+   std::vector<int64_t> dims{static_cast<int64_t>(input.size())};
+ 

